diff -uNdr QLandkarte.2007.03.16.old/src/device/EtrexLegendC/EtrexLegendC.pro QLandkarte.2007.03.16/src/device/EtrexLegendC/EtrexLegendC.pro
--- QLandkarte.2007.03.16.old/src/device/EtrexLegendC/EtrexLegendC.pro	2007-03-04 11:56:33.000000000 +0100
+++ QLandkarte.2007.03.16/src/device/EtrexLegendC/EtrexLegendC.pro	2007-03-23 20:59:27.000000000 +0100
@@ -25,4 +25,4 @@
 QMAKE_POST_LINK = cd ${DESTDIR} && ln -sf ${TARGET} libGPSMap60CS.so
 
 target2.path      = $${QLANDKARTE_LIBDIR}
-target2.commands  = cd $${QLANDKARTE_LIBDIR} && ln -sf ${TARGET} libGPSMap60CS.so
\ No newline at end of file
+target2.commands  = cd ${INSTALL_ROOT}$${QLANDKARTE_LIBDIR} && ln -sf ${TARGET} libGPSMap60CS.so
\ No newline at end of file
diff -uNdr QLandkarte.2007.03.16.old/src/device/GPSMap60CSx/GPSMap60CSx.pro QLandkarte.2007.03.16/src/device/GPSMap60CSx/GPSMap60CSx.pro
--- QLandkarte.2007.03.16.old/src/device/GPSMap60CSx/GPSMap60CSx.pro	2007-03-04 11:56:33.000000000 +0100
+++ QLandkarte.2007.03.16/src/device/GPSMap60CSx/GPSMap60CSx.pro	2007-03-23 21:00:27.000000000 +0100
@@ -23,7 +23,7 @@
 
 QMAKE_POST_LINK   = cd ${DESTDIR} && ln -sf ${TARGET} libGPSMap60Cx.so && ln -sf ${TARGET} libEtrexVentureCx.so
 target2.path      = $${QLANDKARTE_LIBDIR}
-target2.commands  = cd $${QLANDKARTE_LIBDIR} && ln -sf ${TARGET} libGPSMap60Cx.so
+target2.commands  = cd ${INSTALL_ROOT}$${QLANDKARTE_LIBDIR} && ln -sf ${TARGET} libGPSMap60Cx.so
 
 target3.path      = $${QLANDKARTE_LIBDIR}
-target3.commands  = cd $${QLANDKARTE_LIBDIR} && ln -sf ${TARGET} libEtrexVentureCx.so
\ No newline at end of file
+target3.commands  = cd ${INSTALL_ROOT}$${QLANDKARTE_LIBDIR} && ln -sf ${TARGET} libEtrexVentureCx.so
\ No newline at end of file
